### PR TITLE
feat: Set/Get DTR of port

### PR DESCRIPTION
**Pull request by ahollist**
~~This needs to go after #2, as it is based off of it.~~

This lets us fetch the DTR of our port, and set the DTR. 

Testing complete!

### CODE REVIEW

**Review by Russell-Pearce — COMMENTED** [review]
Re-request my review post-testing please

**Comment by ahollist** [comment]
@dsludwig

File: serial/port_linux.go [hunk 1]
```diff
@@ -47,6 +47,35 @@ func (p *Port) SetDeadline(t time.Time) error {
...
+	}
+	if status&unix.TIOCM_DTR > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+// Set the port's DTR pin state
+func (p *Port) SetDTR(state bool) error {
+	// todo(ahollist): Implement
```
Comment (srossross) [inline serial/port_linux.go]: what does this comment mean? this looks implemented

Comment (ahollist) [reply serial/port_linux.go]: Oops, yep that's mean to be removed

Comment (ahollist) [reply serial/port_linux.go]: removed in 4d64207

**Review by Russell-Pearce — APPROVED** [review]
Looks good. Ill approve but few requests foe next time.

Can you make the comments a bit more descriptive? Saying set DTR sets the DTR pin state etc doesnt add much. Just adding "...to let us communicate with pods" or whatever is probably fine.

 Also having setDTR but not using getDTR seems a bit weird to me. Is that a convention?

**Comment by ahollist** [comment]
> Can you make the comments a bit more descriptive? Saying set DTR sets the DTR pin state etc doesnt add much. Just adding "...to let us communicate with pods" or whatever is probably fine.

Yeah, I was planning a documentation PR that would update the whole repo to go-lint conventions, but I can add it in for these methods here first. 
> Also having setDTR but not using getDTR seems a bit weird to me. Is that a convention?

Yeah it is a convention according to [Effective Go](https://golang.org/doc/effective_go.html#Getters), but I personally don't mind continuing to use get/set.

**Comment by Russell-Pearce** [comment]
> Yeah, I was planning a documentation PR that would update the whole repo to go-lint conventions, but I can add it in for these methods here first.
 
Works for me

> Yeah it is a convention according to [Effective Go](https://golang.org/doc/effective_go.html#Getters), but I personally don't mind continuing to use get/set.

That explains it thanks.

**Review by srossross — APPROVED** [review]

**Pull request merged**